content/spin/serverless-ai-api-guide.md

```diff
@@ -53,7 +53,7 @@ The set of operations is common across all supporting language SDKs:
 | Operation | Parameters | Returns | Behavior |
 |:-----|:----------------|:-------|:----------------|
 | `infer`  | model`string`<br /> prompt`string`| `string`  | The `infer` is performed on a specific model.<br /> <br />The name of the model is the first parameter provided (i.e. `llama2-chat`, `codellama-instruct`, or other; passed in as a `string`).<br /> <br />The second parameter is a prompt; passed in as a `string`.<br />|
-| `infer_with_options`  | model`string`<br /> prompt`string`<br /> params`list` | `string`  | The `infer_with_options` is performed on a specific model.<br /> <br />The name of the model is the first parameter provided (i.e. `llama2-chat`, `codellama-instruct`, or other; passed in as a `string`).<br /><br /> The second parameter is a prompt; passed in as a `string`.<br /><br /> The third parameter is a mix of float and unsigned integers relating to inferencing parameters in this order: <br />- `max-tokens` (unsigned 32 integer) Note: the backing implementation may return less tokens. <br /> - `repeat-penalty` (float 32) The amount the model should avoid repeating tokens. <br /> - `repeat-penalty-last-n-token-count` (unsigned 32 integer) The number of tokens the model should apply the repeat penalty to. <br /> - `temperature` (float 32) The randomness with which the next token is selected. <br /> - `top-k` (unsigned 32 integer) The number of possible next tokens the model will choose from. <br /> - `top-p` (float 32) The probability total of next tokens the model will choose from. <br /><br /> The result from `infer_with_options` is a `string` |
+| `infer_with_options`  | model`string`<br /> prompt`string`<br /> params`list` | `string`  | The `infer_with_options` is performed on a specific model.<br /> <br />The name of the model is the first parameter provided (i.e. `llama2-chat`, `codellama-instruct`, or other; passed in as a `string`).<br /><br /> The second parameter is a prompt; passed in as a `string`.<br /><br /> The third parameter is a mix of float and unsigned integers relating to inferencing parameters in this order: <br /><br />- `max-tokens` (unsigned 32 integer) Note: the backing implementation may return less tokens. <br /> Default is 100<br /><br /> - `repeat-penalty` (float 32) The amount the model should avoid repeating tokens. <br /> Default is 1.1<br /><br /> - `repeat-penalty-last-n-token-count` (unsigned 32 integer) The number of tokens the model should apply the repeat penalty to. <br /> Default is 64<br /><br /> - `temperature` (float 32) The randomness with which the next token is selected. <br /> Default is 0.8<br /><br /> - `top-k` (unsigned 32 integer) The number of possible next tokens the model will choose from. <br /> Default is 40<br /><br /> - `top-p` (float 32) The probability total of next tokens the model will choose from. <br /> Default is 0.9<br /><br /> The result from `infer_with_options` is a `string` |
```

Comment:
Looking at the code I would definitely not describe the options record as a "list" - for many programmers that implies a vector or array kind of thing, rather than a record of named values.

I'm also concerned this could be a significant pain to maintain - that's a lot of text and embedded HTML markup squooshed onto one line, easy to get things mixed up!  Instead, I would pull out a separate table called something like "Inference Options" to hold the record info (same content, but now Markdownable), and have the `params` entry and description _refer_ to that table.  E.g.

```suggestion
| `infer_with_options`  | model `string`,<br />prompt `string`,<br />params `record` | `string`  | The `infer_with_options` is performed on a specific model.<br /> <br />The first parameter is the name of the model (e.g. `llama2-chat`, `codellama-instruct`, or other), passed in as a `string`.<br /><br /> The second parameter is a prompt, passed in as a `string`.<br /><br /> The third parameter is a record of options to control how the inferencing is done - see the Inferencing Options table for details. |
```